burnman/mademember.py

```diff
@@ -0,0 +1,274 @@
+# This file is part of BurnMan - a thermoelastic and thermodynamic toolkit for the Earth and Planetary Sciences
+# Copyright (C) 2012 - 2015 by the BurnMan team, released under the GNU
+# GPL v2 or later.
+
+
+from __future__ import absolute_import
+
+import numpy as np
+import warnings
+
+from .mineral import Mineral, material_property
+from .solidsolution import SolidSolution
+from .solutionmodel import *
+from .averaging_schemes import reuss_average_function
+from .processchemistry import sum_formulae
+from . import constants
+
+
+class MadeMember(Mineral):
+
+    """
+    This is the base class for endmembers constructed from a set of other minerals.
+
+    This class is available as :class:`burnman.MadeMember`.
+
+    All the solid solution parameters are expected to be in SI units.  This
+    means that the interaction parameters should be in J/mol, with the T
+    and P derivatives in J/K/mol and m^3/mol.
+    """
+
+    def __init__(self, mineral_list, molar_amounts):
+        """
+        Set up matrices to speed up calculations for when P, T, X is defined.
```

Comment:
Excellent point. Also a couple of the previous lines are no longer relevant after our reworking.